### PR TITLE
Improve Water Pouch

### PR DESCRIPTION
**Pull request by Mahtaran**
Switch to Vanilla methods.
I'll test and add more comments later, but maybe @CrowsOfWar can already take a look at it?

### CODE REVIEW

**Comment by FavouriteDragon** [comment]
Nice!
Hopefully this'll make water pouches less glitchy (for me, they would almost never fill up when I right clicked)!

File: src/main/java/com/crowsofwar/avatar/common/item/ItemWaterPouch.java [hunk 1]
```diff
@@ -62,45 +69,106 @@ public String getModelName(int meta) {
...
+			IBlockState state = world.getBlockState(blockpos);
+;			Material material = state.getMaterial();
+			boolean isCauldron = state.getBlock() instanceof BlockCauldron;
+			// Is the block a cauldron? If yes, set this to the block. If no, set it to null
+			BlockCauldron cauldron = isCauldron ? (BlockCauldron) state.getBlock() : null;
+			if (isCauldron || material == Material.WATER) {
+				// Get how full the block is
+				int level = state.getValue(BlockLiquid.LEVEL).intValue();
+				int canBeFilled;
+				if (isCauldron) {
```
[the remark this answers is not in the document]
Comment (Mahtaran) [reply src/main/java/com/crowsofwar/avatar/common/item/ItemWaterPouch.java]: Yeah, that was what I thought, but then I thought: "nah it's only a few lines". And then it became more and more and more and more.

File: src/main/java/com/crowsofwar/avatar/common/item/ItemWaterPouch.java [hunk 2]
```diff
@@ -62,45 +69,106 @@ public String getModelName(int meta) {
...
+					/* 11 are the flags. Flags are binary. 11 in binary:
+					 *  0 1 0 1 1
+					 * 16 8 4 2 1
+					 * 1: cause block update
+					 * 2: send change to clients
+					 * 4: prevent re-render
+					 * 8: run re-renders on main thread
+					 * 16: prevent observers update
+					 * For more information see {@link net.minecraft.world.World#setBlockState World#setBlockState}
+					 */
```
[the remark this answers is not in the document]
Comment (Mahtaran) [reply src/main/java/com/crowsofwar/avatar/common/item/ItemWaterPouch.java]: Oh, I thought the random number would be confusing to people reading through the code, I can remove it if you want

**Comment by Mahtaran** [comment]
@CrowsOfWar okay it works now
Can you review it please?

**Pull request merged**